src/frame.js

```diff
@@ -0,0 +1,55 @@
+const STRIKE = 10;
+const STANDARD_ROUND = 2;
+const STRIKE_ROUND = 1;
+const STRIKE_SCORE_ROUND = 3;
+
+class Frame {
+  constructor(strikeRound = STRIKE_ROUND, spareRound = STANDARD_ROUND) {
+    this.rounds = [];
+    this.rules = {
```

Comment:
I like how you've turned the rules into a dictionary - that's known as a Strategy pattern and it's good practice with this kind of logic.

I would suggest however that you don't fall into the trap of treating all rolls the same. A roll may just be a number programmatically, but a first or second roll is different from a bonus roll in *purpose*. We should always think about what a things represents, not what represents the thing, and in this case I would suggest that if you treat rolls and bonuses and different things you'll end up with an even cleaner implementation.